aws-blog-emrencyrption/cfn/kms.json

```diff
@@ -0,0 +1,240 @@
+{
+    "AWSTemplateFormatVersion": "2010-09-09",
+    "Description": "Generate KMS keys for EMRFS & LUKS Encryption",
+    "Outputs": {
+        "EMRKey": {
+            "Description": "KMS Key for EMRFS",
+            "Value": {
+                "Fn::GetAtt": [
+                    "KMSEMRKey",
+                    "Arn"
+                ]
+            }
+        },
+        "LUKSKey": {
+            "Description": "KMS Key for LUKS",
+            "Value": {
+                "Fn::GetAtt": [
+                    "KMSLUKSKey",
+                    "Arn"
+                ]
+            }
+        }
+    },
+    "Parameters": {
+        "KMSEMRAlias": {
+            "Default": "alias/EMRFS-Encryption-Key",
+            "Description": "KMS alias to be used for EMRFS",
+            "Type": "String"
+        },
+        "KMSLUKSAlias": {
+            "Default": "alias/LUKS-Encryption-Key",
+            "Description": "KMS alias to be used for LUKS Encryption",
+            "Type": "String"
+        }
+    },
+    "Resources": {
+        "EMRAlias": {
+            "Properties": {
+                "AliasName": {
+                    "Ref": "KMSEMRAlias"
+                },
+                "TargetKeyId": {
+                    "Ref": "KMSEMRKey"
+                }
+            },
+            "Type": "AWS::KMS::Alias"
+        },
+        "KMSEMRKey": {
+            "Properties": {
+                "Description": "Master Key that will be used for EMR Encryption",
+                "Enabled": "true",
+                "KeyPolicy": {
+                    "Statement": [
+                        {
+                            "Action": [
+                                "kms:Encrypt",
+                                "kms:Decrypt",
+                                "kms:ReEncrypt",
+                                "kms:GenerateDataKey",
+                                "kms:DescribeKey"
```

Comment:
For EMRFS encryption, the permission that we need are:
"kms:Encrypt",
"kms:Decrypt",
"kms:GenerateDataKey",